### PR TITLE
Kairosdb tests

### PR DESCRIPTION
**Pull request by ch40s3k**
Adding tests to kairosdbhandler, new tag parsing in collector

### CODE REVIEW

File: src/diamond/handler/kairosdb.py [hunk 1]
```diff
@@ -2,22 +2,39 @@
 Send metrics to a [KairosDB](http://kairosdb.github.io/) using telnet or rest
 method.
 
+add config to diamond.conf:
```
Comment (deejay1) [inline src/diamond/handler/kairosdb.py]: _Add the folowing configuration_

File: src/diamond/handler/kairosdb.py [hunk 2]
```diff
@@ -2,22 +2,39 @@
...
+
+[[KairosDBHandler]]
+host = localhost
+port = 4242
+
+Optionaly if you like to usa Tags (Metrics 2.0), you should add field [[[tags]]] in collector config.
```
Comment (deejay1) [inline src/diamond/handler/kairosdb.py]: drop the metrics 2.0 part

File: src/diamond/handler/kairosdb.py [hunk 3]
```diff
@@ -2,22 +2,39 @@
...
+Example:
+
+[[CPUCollector]]
+enabled = True
+[[[tags]]]
+env=develop
+host=localhost
```
Comment (deejay1) [inline src/diamond/handler/kairosdb.py]: drop the host please, as to not confuse people

File: src/diamond/handler/kairosdb.py [hunk 4]
```diff
@@ -2,22 +2,39 @@
...
 from Handler import Handler
+from diamond.collector import get_hostname
 import socket
 
 
 class KairosDBHandler(Handler):
     """
-    Implements the abstract Handler class, sending data to kairosdb
+    Implements the abstract Handler class, sending data to kairosdb.
+    It sets by default tag [host] which indicates hostname.
```
Comment (deejay1) [inline src/diamond/handler/kairosdb.py]: the default tag `host`

File: src/diamond/handler/test/testkairosdbhandler.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        patch_send.start()
+        handler.process(metric)
+        patch_send.stop()
+        patch_sock.stop()
+        self.assertEqual(sendmock.call_count, 1)
+        self.assertEqual(sendmock.call_args_list[0], expected_data)
+
+
+if __name__ == "__main__":
+    unittest.main()
```
Comment (deejay1) [inline src/diamond/handler/test/testkairosdbhandler.py]: missing newline at the end

File: src/diamond/handler/kairosdb.py (hunk 1 above)
Comment (ch40s3k) [inline src/diamond/handler/kairosdb.py]: Done

File: src/diamond/handler/kairosdb.py (hunk 2 above)
Comment (ch40s3k) [inline src/diamond/handler/kairosdb.py]: Done

File: src/diamond/handler/test/testkairosdbhandler.py (hunk 5 above)
Comment (ch40s3k) [inline src/diamond/handler/test/testkairosdbhandler.py]: Done

File: src/diamond/handler/kairosdb.py (hunk 4 above)
Comment (ch40s3k) [inline src/diamond/handler/kairosdb.py]: Done

File: src/diamond/handler/kairosdb.py (hunk 3 above)
Comment (ch40s3k) [inline src/diamond/handler/kairosdb.py]: Done

File: src/diamond/handler/kairosdb.py [hunk 6]
```diff
@@ -2,22 +2,39 @@
 Send metrics to a [KairosDB](http://kairosdb.github.io/) using telnet or rest
 method.
 
+Add the following configuration to diamond.conf:
+
+[[KairosDBHandler]]
+host = localhost
+port = 4242
+
+Optionaly if you like to usa Tags, you should add field [[[tags]]] in collector config.
```
Comment (deejay1) [inline src/diamond/handler/kairosdb.py]: use, not usa ;)

Comment (ch40s3k) [inline src/diamond/handler/kairosdb.py]: bleh, i saw that earlier and totaly forget ;/, Done

File: src/diamond/collector.py [hunk 7]
```diff
@@ -254,6 +257,8 @@ def get_default_config_help(self):
                                  'Mutually exclusive with metrics_blacklist',
             'metrics_blacklist': 'Regex to match metrics to block. ' +
                                  'Mutually exclusive with metrics_whitelist',
+            'tags': '(optional) space seperated "key1=value1 key2=value2" tags'
+            + ' to be attached to the metrics',
```
Comment (h00dy) [inline src/diamond/collector.py:261]: This is not correct right now.

**Comment by deejay1** [comment]
Also merge conflicts, seems like you messed up with the rebase ;)

**Comment by deejay1** [comment]
@ch40s3k Can you fix the conflicts?

**Comment by ch40s3k** [comment]
sure, np

File: src/diamond/collector.py [hunk 8]
```diff
@@ -254,6 +257,8 @@ def get_default_config_help(self):
                                  'Mutually exclusive with metrics_blacklist',
             'metrics_blacklist': 'Regex to match metrics to block. ' +
                                  'Mutually exclusive with metrics_whitelist',
+            'tags': '(optional) tags in format [[[tags]]]'
+            + ' to be attached to the metrics',
```
Comment (h00dy) [inline src/diamond/collector.py:261]: This message is still wrong.
It says that in config I can write
tags: [[[some tags]]]

**Pull request closed without merging**